js/ui/text_box/text_box.js

```diff
@@ -97,16 +97,13 @@ const TextBox = TextEditor.inherit({
 
 
     _renderLabel: function() {
-        this.callBase();
-
-        if(this._$searchIcon && this._$label) {
-            const labelBeforeElement = this._$label.find('.dx-label-before');
-            const labelBeforeWidth = getWidth(labelBeforeElement) + getOuterWidth(this._$searchIcon);
+        const options = {
+            editor: this,
+            container: this._input(),
+            icon: this._$searchIcon
+        };
```

Comment:
```suggestion
        const options = {
            editor: this,
            container: this._input(),
            icon: this._$searchIcon
        };
```

Let's extract it to a separate method